src/ts/pausedtimeshiftupdatehandler.ts

```diff
@@ -0,0 +1,58 @@
+import { PlayerAPI } from 'bitmovin-player';
+import { Timeout } from './timeout';
+
+export class PausedTimeshiftUpdatedHandler {
+  private static instance: PausedTimeshiftUpdatedHandler;
+  private static _player: PlayerAPI;
+  private pausedTimeshiftUpdater: Timeout;
+  private handlers: CallableFunction[] = [];
+
+  private constructor() {}
+
+  public static getInstance(player: PlayerAPI) {
+    if (!this.instance) {
+      this._player = player;
+      this.instance = new PausedTimeshiftUpdatedHandler();
+    }
+
+    return this.instance;
+  }
```

Comment:
We can't do that 🤔 What if there are two player instances on one page (two UI instances). As they might play different content we would have the same handler (same player) 🤔

Reply:
Correct, reverted.